uniffi/src/cesride.udl

```diff
@@ -0,0 +1,117 @@
+namespace cesride {
+};
+
+[Error]
+enum CESRError {
+    "General",
+    "Matter",
+    "EmptyMaterial",
+    "Decode",
+    "UnexpectedCode",
+    "UnexpectedCountCode",
+    "UnexpectedOpCode",
+    "InvalidVarSize",
+    "InvalidVarRawSize",
+    "InvalidVarIndex",
+    "InvalidCodeSize",
+    "InvalidBase64Character",
+    "InvalidBase64Index",
+    "Shortage",
+    "EmptyQb64",
+    "UnknownSizage",
+    "UnknownHardage",
+    "UnknownBardage",
+    "UnsupportedSize",
+    "TooSmall",
+    "Prepad",
+    "NonZeroedPrepad",
+    "NonZeroedLeadByte",
+    "NonZeroedLeadBytes",
+    "NonZeroedPadBits",
+    "Parsing",
+    "ParseQb64",
+    "ParseQb2",
+    "Conversion",
+    "Value",
+    "Validation",
+};
+
+interface Cigar {
+    [Throws=CESRError]
+    constructor(
+        Verfer verfer,
+        optional string? code = null,
+        optional sequence<u8>? raw = null,
+        optional string? qb64 = null,
+        optional sequence<u8>? qb64b = null,
+        optional sequence<u8>? qb2 = null
+    );
+
+    // Verfer verfer();
+    // void set_verfer([ByRef] Verfer verfer);
+    sequence<u8> raw();
+    string code();
+};
+
+interface Dater {
```

Comment:
Defining `Dater` (and other primitives) as an `interface` cause adding complexity when we write bindings for groups that are defined as Vec<some primitive>. Uniffi forces you to use `Arc` smart pointers in this case. 
If we define everything as `dictionaries` this issue go away but we need to define and expose extra functions like here: https://github.com/WebOfTrust/parside/pull/2/files#diff-96acc651ada67f2389ec586f0b6292f8c92b1ab2fbb08ed43c865f15133a9f1aR4